napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,288 @@
+            elem_name="color",
+            default="white",
+        )
+        broadcasted_colors = normalize_and_broadcast_colors(
+            n_colors, transformed_color
+        )
+        self._values = np.concatenate((self.values, broadcasted_colors))
+
+    def remove(self, indices_to_remove: Union[set, list, np.ndarray]):
+        """Remove the indicated color elements
+
+        Parameters
+        ----------
+        indices_to_remove : set, list, np.ndarray
+            The indices of the text elements to remove.
+        """
+        selected_indices = list(indices_to_remove)
+        if len(selected_indices) > 0:
+            self._values = np.delete(self.values, selected_indices, axis=0)
+
+    def refresh_colors(
+        self, properties: dict, update_color_mapping: bool = False
+    ):
+        """Calculate and update face or edge colors if using a cycle or color map
+
+        Parameters
+        ----------
+        properties : dict
+            The layer properties to map the colors against.
+        update_color_mapping : bool
+            If set to True, the function will recalculate the color cycle map
+            or colormap (whichever is being used). If set to False, the function
+            will use the current color cycle map or color map. For example, if you
+            are adding/modifying points and want them to be colored with the same
+            mapping as the other points (i.e., the new points shouldn't affect
+            the color cycle map or colormap), set update_color_mapping=False.
+            Default value is False.
+        """
+        if self._mode in [ColorMode.CYCLE, ColorMode.COLORMAP]:
+            if self._mode == ColorMode.CYCLE:
+                color_properties = properties[self.color_property]
+                colors = self.categorical_colormap.map(color_properties)
+
+            elif self._mode == ColorMode.COLORMAP:
+                color_properties = properties[self.color_property]
+                if len(color_properties) > 0:
+                    if (
+                        update_color_mapping
+                        or self.continuous_contrast_limits is None
+                    ):
+                        colors, contrast_limits = map_property(
+                            prop=color_properties,
+                            colormap=self.continuous_colormap,
+                        )
+                        self.continuous_contrast_limits = contrast_limits
+                    else:
+                        colors, _ = map_property(
+                            prop=color_properties,
+                            colormap=self.continuous_colormap,
+                            contrast_limits=self.continuous_contrast_limits,
+                        )
+                else:
+                    colors = np.empty((0, 4))
+            if len(colors) == 0:
+                colors = np.empty((0, 4))
+            self.values = colors
+            self.events.values()
```

Comment:
we shouldn't need to do the manual event emission. Setting the `values` attribute should trigger the event via the evented dataclass